packages/features/tasker/tasks/sendEmail.ts

```diff
@@ -8,9 +8,12 @@ const sendEmailPayloadSchema = z.object({
   payload: z.string(),
 });
 
+type SendEmailPayload = z.infer<typeof sendEmailPayloadSchema>;
+
 export async function sendEmail(payload: string): Promise<void> {
   try {
-    const parsedPayload = sendEmailPayloadSchema.parse(JSON.parse(payload));
+    const parsedPayloadObj: SendEmailPayload = JSON.parse(payload);
```

Comment:
So now if the string payload we pass to this function doesn't match the `SendEmailPayload` type that we have inferred from the zod schema defined earlier, we would get a `TypeError` exception, correct?

@anikdhabal Please help me out in understanding this aspect of the changes

Reply:
Payload is saved in DB as string always. So no. This won't prevent getting an error on compile time. We also need to keep the runtime protection as well.

Reply:
Got it @zomars. Will have a further look